src/lib.rs

```diff
@@ -99,33 +124,67 @@ pub fn run(args: Args) {
             );
 
             get_searcher(output_mode)
+                .borrow_mut()
                 .search_path(&matcher, path, printer.sink_with_path(&matcher, path))
                 .unwrap();
+            buffer_writer.print(printer.get_mut()).unwrap();
```

Comment:
I think what's happening here (+ in `get_printer()`) is that the `BufferWriter` hands out `Buffer`'s (which serve as each printer's inner `Writer`) but then here there's just the single `BufferWriter` (across all threads) and it's getting passed a reference to this printer's `Buffer` (via `.get_mut()`) in order to then actually print out its contents?